Small improvements

BlockyTheDev opened the pull request.

BlockyTheDev committed:
Fix SuppressWarnings for deprecation

BlockyTheDev committed:
Remove unused methods and parameters

BlockyTheDev committed:
Correct version variable for version 1.19.1

BlockyTheDev committed:
Remove not needed empty String concatenation

RodneyMKay commented on src/main/java/de/redgames/f3nperm/ServerVersion.java:
```diff
@@ -9,7 +9,7 @@ public final class ServerVersion {
     public static final ServerVersion v_1_18 = new ServerVersion(1, 18, 0);
     public static final ServerVersion v_1_18_2 = new ServerVersion(1, 18, 2);
-    public static final ServerVersion v_1_19 = new ServerVersion(1, 19, 1);
+    public static final ServerVersion v_1_19_1 = new ServerVersion(1, 19, 1);
```
If I'm not mistaken, spigot 1.19(.0) NMS was internally labelled as V1_19_R1, which is why it's labelled accordingly here. A comment which documents that probably wouldn't hurt though.

BlockyTheDev replied:
> If I'm not mistaken, spigot 1.19(.0) NMS was internally labelled as V1_19_R1, which is why it's labelled accordingly here. A comment which documents that probably wouldn't hurt though.

Sorry, I didn't think of NMS names because of the ServerVersion, with the three numbers (major, minor, patch), as I normally use the Server Version for naming variables in such cases... I will change it back and name the `1.19.4` accordingly to NMS names.

BlockyTheDev replied:
> A comment which documents that probably wouldn't hurt though.

I renamed the class to `NMSVersion` because I think it is better to name it as what it includes. When I first read "ServerVersion" I think it is something like `1.19.2`, `1.19.3`, `1.19.4`, ... Also I changes the variable names to better show what NMS version it is because the package is named the same way (`org.bukkit.craftbukkit.v1_19_R3`). In addition I changed the server versions for `v1_17_R1` and `v1_18_R1` because the NMS version `v1_17_R0`/`v1_18_R0` is not existing, now it is equal to the NMS version syntax like the `v1_19_R1` one.

BlockyTheDev committed:
Correct my mistake and give class a better name

BlockyTheDev commented:
> A comment which documents that probably wouldn't hurt though.

I renamed the class to `NMSVersion` because I think it is better to name it as what it includes. When I first read "ServerVersion" I think it is something like `1.19.2`, `1.19.3`, `1.19.4`, ... Also I changes the variable names to better show what NMS version it is because the package is named the same way (`org.bukkit.craftbukkit.v1_19_R3`). In addition I changed the server versions for `v1_17_R1` and `v1_18_R1` because the NMS version `v1_17_R0`/`v1_18_R0` is not existing, now it is equal to the NMS version syntax like the `v1_19_R1` one.

BlockyTheDev committed:
Rename provider classes

RodneyMKay commented:
Yea, that's a good idea! Thanks for all your contributions! It's greatly appreciated :D
